### PR TITLE
pranay/VAN 3196

### PR DESCRIPTION
**Pull request by pranaySinghDev**
VAN-3196 
Azure CICD Modules Added
Updated CI Modules Sonatype nexus, SonarQube, Fortify, dotnet build and publish templates 
Added Azure function deployment template

### CODE REVIEW

File: pipeline-modules/app-service/azure/fortify.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          curl -X POST "https://ssc.nxplatform.io/ssc/api/v1/projectVersions/${id}/artifacts" -H "accept: application/json" -H "Authorization: FortifyToken $CPI__FORTIFY_TOKEN" -H "Content-Type: multipart/form-data" -F "file=@worker.fpr"
+          curl -k -X GET "https://ssc.nxplatform.io/ssc/api/v1/projectVersions/${id}/issues?start=0&limit=200&fields=friority&showhidden=false&showremoved=false&showsuppressed=false&showshortfilenames=false" -H "accept: application/json" -H "Authorization: FortifyToken $CPI__FORTIFY_TOKEN" > $(Build.BinariesDirectory)/out/output.log
+          cat $(Build.BinariesDirectory)/out/output.log
+          critical_count=$(cat $(Build.BinariesDirectory)/out/output.log | jq '.data[].friority' | grep "Critical" | wc -l)
+          if [ $critical_count -gt 0 ]; then
+            # Print an error message andset pipeline fail variable
+            echo "Error: There are 'critical' severity vulnerabilities in the code."
+            failPipeline=false
+          fi
+          high_count=$(cat $(Build.BinariesDirectory)/out/output.log | jq '.data[].friority' | grep "High" | wc -l)
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/fortify.yaml:57]: should "friority" be "priority"?

Comment (dshelley66) [reply pipeline-modules/app-service/azure/fortify.yaml:57]: ah I see from the Fortify doc that "friority" is a thing :)

File: pipeline-modules/app-service/azure/pipeline-config.yaml [hunk 2]
```diff
@@ -52,7 +52,7 @@ template: |
   {% if pool_name %}
     name: {{ pool_name }}
   {% else %}
-    vmImage: ubuntu-latest
+    vmImage: AppSec Pool
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/pipeline-config.yaml:55]: Is this specific to RSAF or to Azure?

File: pipeline-modules/deployment-service/azure/pipeline-config.yaml [hunk 3]
```diff
@@ -71,6 +88,12 @@ template: |
   name: '$(Date:yyyyMMdd)$(Rev:.r)'
   trigger: none
   steps:
+  - script: |
+      rm -rf ~/.gitconfig 
+      git config --global url."https://$(git_username):$(GIT_PASSWORD)@github.com/".insteadOf "https://github.com/"
+      cat ~/.gitconfig
+      echo $(GIT_PASSWORD)
```
Comment (dshelley66) [inline pipeline-modules/deployment-service/azure/pipeline-config.yaml:95]: we shouldn't leave this in or the cat .gitconfig

File: pipeline-modules/app-service/azure/sonatype-nexus.yaml [hunk 4]
```diff
@@ -3,32 +3,41 @@ name: sonatype-nexus-iq
...
+      type: string
+      default: ""
+    pool_name:
+      title: Pool Name
+      description: Use selfhosted agent by proving agent pool name.
+      type: string
+      default: ""
+    applicationName:
+      title: Application name
+      description: Codepipes application name
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/sonatype-nexus.yaml:31]: Should be Code Pipes

File: pipeline-modules/app-service/azure/dotnet-build.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      description: Use selfhosted agent by proving agent pool name.
+      type: string
+      default: ""
+    working_dir:
+      title: Code Directory Path
+      description: Directory Path where the code is located.
+      type: string
+    applicationName:
+      title: Application name
+      description: Codepipes application name
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/dotnet-build.yaml:27]: Code Pipes

File: pipeline-modules/app-service/azure/fortify.yaml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+inputs:
+  properties:
+    pool_name:
+      title: Pool Name
+      description: Use selfhosted agent by proving agent pool name.
+      type: string
+      default: ""
+    fortify_application_id:
+      title: Fortify Application ID
+      description: If not provided codepipes application name will be used
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/fortify.yaml:27]: Code Pipes

File: pipeline-modules/app-service/azure/fortify.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      type: string
+      default: ""
+    applicationName:
+      title: Application name
+      description: Codepipes application name
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/fortify.yaml:32]: Code Pipes

File: pipeline-modules/app-service/azure/publish-artifact.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+inputs:
+  properties:
+    pool_name:
+      title: Pool Name
+      description: Use selfhosted agent by proving agent pool name.
+      type: string
+      default: ""
+    applicationName:
+      title: Application name
+      description: Codepipes application name
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/publish-artifact.yaml:23]: Code Pipes

File: pipeline-modules/app-service/azure/rsaf-appsec.yaml [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+inputs:
+  properties:
+    git_local_path:
+      title: Local Clone Directory
+      description: The name of a new directory to clone into.
+      type: string
+      default: repo
+    applicationName:
+      title: Application name
+      description: Codepipes application name
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/rsaf-appsec.yaml:24]: Code Pipes

File: pipeline-modules/app-service/azure/sonarqube.yaml [hunk 10]
```diff
@@ -7,66 +7,61 @@ description: Analyze source code using Sonar Qube
...
+env:
+  - CPI__SONARQUBE_SERVICE_CONNECTION
 inputs:
   properties:
-    git_checkout_target:
-      title: Checkout Target
-      description: Git branch, tag or commit hash to checkout.
+    applicationName:
+      title: Application name
+      description: Codepipes application name
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/sonarqube.yaml:20]: Find anywhere that you have "codepipes" or "Codepipes" in a description and change to Code Pipes

File: pipeline-modules/app-service/azure/rsaf-appsec.yaml [hunk 11]
```diff
@@ -0,0 +1,137 @@
+provisioner: azure
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/rsaf-appsec.yaml:1]: I guess we didn't circle back on the discussion about private repo for customer specific stuff....not sure it makes sense to put a file with this name into this repo?

**Comment by dshelley66** [comment]
I assume this will be squashed before merge?

File: pipeline-modules/deployment-service/azure/pipeline-config.yaml (hunk 3 above)
Comment (pranaySinghDev) [reply pipeline-modules/deployment-service/azure/pipeline-config.yaml:95]: removed

File: pipeline-modules/app-service/azure/rsaf-appsec.yaml (hunk 11 above)
Comment (pranaySinghDev) [reply pipeline-modules/app-service/azure/rsaf-appsec.yaml:1]: i'll change the file name too, i did for description.

File: pipeline-modules/infra-service/azure/terraform.yaml [hunk 12]
```diff
@@ -206,4 +206,5 @@ template: |
       {{ upload_artifact(tf_state_file) }}  {# upload the state even if the TF action failed as it may have already been modified #}
       {% endif %}
       az logout
+    timeoutInMinutes: 300
```
Comment (dshelley66) [inline pipeline-modules/infra-service/azure/terraform.yaml:209]: what is this timeout for? Terraform apply/destroy operations can easily take longer than 5 minutes....

Comment (shwetanshu-cldcvr) [reply pipeline-modules/infra-service/azure/terraform.yaml:209]: The timeout is in minutes to 300 mins.

**Review by shwetanshu-cldcvr — APPROVED** [review]
LGTM

**Pull request merged**